cloudinit/sources/DataSourceNoCloud.py

```diff
@@ -362,18 +363,50 @@ def _merge_new_seed(cur, seeded):
     return ret
 
 
+def _parse_dmi() -> str:
+    serial = dmi.read_dmi_data("product_serial")
```

Comment:
s/product_serial/system-serial-number/ I think, otherwise this always fails

Reply:
@rvandegrift It looks like applying this actually broke the test case, what were you seeing that led to this suggestion?

Reply:
@holmanb , I think `system-serial-number` is "more right" here. We use the map @rvandegrift mentioned to read the DMI value from `/sys`. If that fails, we fallback to reading using `dmidecode`. It looks like specifying `product_serial` as a key will trigger the fallback code, but we should be using `/sys` if it's available.

> broke the test case

Which test case? Mine will be broken unless you apply my other suggested fix :grimacing: